extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ClientFactory.java

```diff
@@ -138,6 +219,14 @@ private static SdkAsyncHttpClient getOrBuildHttpAsyncClient(@NotNull final S3Ins
                 .build());
     }
 
+    private static SdkHttpClient buildHttpSyncClient(@NotNull final S3Instructions instructions) {
+        // TODO: cache?
```

Comment:
I think we can enable caching since the client object internally maintains a pool of connections, and keeps on reconnecting as required.
Also, can you check if this HttpClient object will be closed, because then we might not be able to cache it.

Reply:
I've updated this to add it to a cache; but I'm not sure exactly what your question is.

Reply:
What I meant was that when the `S3Client` is closed, will it also close the underlying `SdkHttpClient`?

But I just read the comments in `SdkSyncClientBuilder::httpClient`, that the client won't be closed if its being set externally rather than being built by using the `SdkSyncClientBuilder::httpClientBuilder`.

So its good to cache it.